homeassistant/components/suez_water/sensor.py

```diff
@@ -51,45 +47,30 @@ def __init__(self, client: SuezClient, counter_id: int) -> None:
             manufacturer="Suez",
         )
 
-    def _fetch_data(self) -> None:
-        """Fetch latest data from Suez."""
-        try:
-            self.client.update()
-            # _state holds the volume of consumed water during previous day
-            self._attr_native_value = self.client.state
-            self._attr_available = True
-            self._attr_attribution = self.client.attributes["attribution"]
-
-            self._attr_extra_state_attributes["this_month_consumption"] = {}
-            for item in self.client.attributes["thisMonthConsumption"]:
-                self._attr_extra_state_attributes["this_month_consumption"][item] = (
-                    self.client.attributes["thisMonthConsumption"][item]
-                )
-            self._attr_extra_state_attributes["previous_month_consumption"] = {}
-            for item in self.client.attributes["previousMonthConsumption"]:
-                self._attr_extra_state_attributes["previous_month_consumption"][
-                    item
-                ] = self.client.attributes["previousMonthConsumption"][item]
-            self._attr_extra_state_attributes["highest_monthly_consumption"] = (
-                self.client.attributes["highestMonthlyConsumption"]
-            )
-            self._attr_extra_state_attributes["last_year_overall"] = (
-                self.client.attributes["lastYearOverAll"]
-            )
-            self._attr_extra_state_attributes["this_year_overall"] = (
-                self.client.attributes["thisYearOverAll"]
-            )
-            self._attr_extra_state_attributes["history"] = {}
-            for item in self.client.attributes["history"]:
-                self._attr_extra_state_attributes["history"][item] = (
-                    self.client.attributes["history"][item]
-                )
-
-        except PySuezError:
-            self._attr_available = False
-            _LOGGER.warning("Unable to fetch data")
-
-    def update(self) -> None:
-        """Return the latest collected data from Suez."""
-        self._fetch_data()
-        _LOGGER.debug("Suez data state is: %s", self.native_value)
+    @property
+    def native_value(self) -> float | None:
+        """Return the current daily usage."""
+        if self.coordinator.aggregated_data is None:
+            return None
+        return self.coordinator.aggregated_data.value
+
+    @property
+    def attribution(self) -> None | str:
+        """Return data attribution message."""
+        if self.coordinator.aggregated_data is None:
+            return None
+        return self.coordinator.aggregated_data.attribution
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return aggregated data."""
+        if self.coordinator.aggregated_data is None:
+            return None
+        return {
```

Comment:
Okay so for a general comment. We want to minimize extra state attribute usage. They are also stored in the state and that can slow the system down if that happens enough. So ideally, we use service actions in the future to fetch the large objects (like the object with the consumption this month) and we create individual entities out of variables like `this year overall`. Doesn't have to happen in this PR or tomorrow, but please put this on your todo list for improvements

Reply:
Yes I totally agree with you but this was the data retrieved by the previous owner before I made any changes.

The other changes I want to make to the integration are meant to address this problem.